agent/util-scripts/pbench-register-tool

```diff
@@ -1,184 +1,343 @@
+			/bin/rm -f "${this_label_file}"
+			echo "${local_label}" > "${this_label_file}"
 		fi
 	else
-		error_log "Failed to install tool $name ($pbench_bin/tool-scripts/$name --install \"${tool_opts[@]}\")\n"
+		error_log "Failed to install tool ${name} (${pbench_bin}/tool-scripts/${name} --install \"${tool_opts[@]}\")\n"
 	fi
+fi
+
+# Make sure we capture any local error codes to report failure later of the
+# overall script.
+exit_rc=${rc}
+
+if [[ ${do_install} == 1 ]]; then
+	install_opt=""
 else
-	# register this tool on the remote host
-	if [ -z "$label" ]; then
+	isntall_opt="--no-install"
+fi
+for remote in ${remotes[@]}; do
+	# Register this tool on a remote host
+	label=${labels_A[${remote}]}
+	if [[ -z "${label}" ]]; then
 		label_opt=""
 	else
-		label_opt="--label=$label"
+		label_opt=" --label=${label}"
 	fi
-	# can't really pass an array of tool options over ssh, so here we will actually construct the array within the ssh call, then reference that array
-	debug_log "$tool_opt_array_cmd"' pbench-register-tool --name='"$name"' --group='"$group"' $label_opt -- "${tool_opts[@]}" 2>&1'
-	ssh $ssh_opts $remote "$tool_opt_array_cmd"' pbench-register-tool --name='"$name"' --group='"$group" $label_opt' -- "${tool_opts[@]}" 2>&1' | sed -e 's/\(.*\)/'[$remote]'\1/g'
-	rc=$?
-	if [ "$rc" != "0" ]; then
-		error_log "Failed to register tool $name remotely\n"
+	# Can't really pass an array of tool options over ssh, so here we will
+	# actually construct the array within the ssh call, then reference that
+	# array.
+	debug_log "${tool_opt_array_cmd}"' pbench-register-tool --name='"${name}"' --group='"${group}"${install_opt}${label_opt}' -- "${tool_opts[@]}" 2>&1'
+	ssh ${ssh_opts} ${remote} "${tool_opt_array_cmd}"' pbench-register-tool --name='"${name}"' --group='"${group}"${install_opt}${label_opt}' -- "${tool_opts[@]}" 2>&1' | sed -e 's/\(.*\)/'[${remote}]'\1/g'
+	rc=${?}
+	let exit_rc=${exit_rc}+${rc}
+	if [[ ${rc} -ne 0 ]]; then
+		error_log "Failed to register tool ${name} on remote host ${remote}\n"
+		continue
 	fi
-	if [ ! -d "$pbench_run/tools-$group" ]; then
-		mkdir -p "$pbench_run/tools-$group"
+	if [[ ! -d "${pbench_run}/tools-${group}" ]]; then
+		mkdir "${pbench_run}/tools-${group}"
+		if [[ ${?} -ne 0 ]]; then
+			error_log "Unable to create the necessary tools group directory, ${pbench_run}/tools-${group}"
+			exit 1
+		fi
 	fi
-	this_tool_file="$pbench_run/tools-$group/remote@$remote"
-	/bin/rm -f "$this_tool_file"
-	# the specific tool options are stored on the remote host, so there is no need to keep them here
-	# however, we do use this file to store the label for this remote host [if any]
-	if [ -z "$label" ]; then
-		touch "$this_tool_file"
+	this_tool_file="${pbench_run}/tools-${group}/remote@${remote}"
+	/bin/rm -f "${this_tool_file}"
+	# The specific tool options are stored on the remote host, so there is
+	# no need to keep them here however, we do use this file to store the
```

Comment:
Misplaced comma: "... keep them here, however we do use..."

Reply:
Thanks, fixed.